crates/rspack_plugin_devtool/src/lib.rs

```diff
@@ -112,11 +112,14 @@ impl Plugin for DevtoolPlugin {
           map.set_file(Some(filename.clone()));
           for source in map.sources_mut() {
             let uri = normalize_custom_filename(source);
-            let resource_path = if let Some(relative_path) = diff_paths(uri, &context) {
+            let mut resource_path = if let Some(relative_path) = diff_paths(uri, &context) {
               relative_path.to_string_lossy().to_string()
             } else {
               uri.to_owned()
             };
+            if cfg!(target_os = "windows") {
```

Comment:
seems weird to me, how webpack handle this problem?

Reply:
const _contextify = (context, request) => {
	return request
		.split("!")
		.map(r => absoluteToRequest(context, r))
		.join("!");
webpack is in here

Reply:
can you paste the link

Reply:
const WINDOWS_ABS_PATH_REGEXP = /^[a-zA-Z]:[\\/]/;
const SEGMENTS_SPLIT_REGEXP = /([|!])/;
const WINDOWS_PATH_SEPARATOR_REGEXP = /\\/g;
const absoluteToRequest = (context, maybeAbsolutePath) => {
	if (maybeAbsolutePath[0] === "/") {
		if (
			maybeAbsolutePath.length > 1 &&
			maybeAbsolutePath[maybeAbsolutePath.length - 1] === "/"
		) {
			// this 'path' is actually a regexp generated by dynamic requires.
			// Don't treat it as an absolute path.
			return maybeAbsolutePath;
		}

		const querySplitPos = maybeAbsolutePath.indexOf("?");
		let resource =
			querySplitPos === -1
				? maybeAbsolutePath
				: maybeAbsolutePath.slice(0, querySplitPos);
		resource = relativePathToRequest(path.posix.relative(context, resource));
		return querySplitPos === -1
			? resource
			: resource + maybeAbsolutePath.slice(querySplitPos);
	}

	if (WINDOWS_ABS_PATH_REGEXP.test(maybeAbsolutePath)) {
		const querySplitPos = maybeAbsolutePath.indexOf("?");
		let resource =
			querySplitPos === -1
				? maybeAbsolutePath
				: maybeAbsolutePath.slice(0, querySplitPos);
		resource = path.win32.relative(context, resource);
		if (!WINDOWS_ABS_PATH_REGEXP.test(resource)) {
			resource = relativePathToRequest(
				resource.replace(WINDOWS_PATH_SEPARATOR_REGEXP, "/")
			);
		}
		return querySplitPos === -1
			? resource
			: resource + maybeAbsolutePath.slice(querySplitPos);
	}

	// not an absolute path
	return maybeAbsolutePath;
};

Reply:
He did it and I see we have this method in the warehouse, let me see how to fight it with him

Reply:
let source_path = contextify(&context, &source);
            let uri = normalize_custom_filename(&source_path);
            let resource_path = if let Some(relative_path) = diff_paths(uri, &context) {
              relative_path.to_string_lossy().to_string()
            } else {
              uri.to_owned()
            };
I think this implementation should work